File: liveroom-uikit/src/main/java/com/qlive/uikit/RoomListPage.java
package com.qlive.uikit;

import com.qlive.sdk.QPage;

/**
 * 房间列表页面
 */
public class RoomListPage implements QPage {
    private int customLayoutID = -1;

    public final int getCustomLayoutID() {
        return this.customLayoutID;
    }

    /**
     * 设置房间列表页面的自定义布局
     * @param layoutID  拷贝kit_activity_room_list.xml 修改后的自定义布局
     */
    public final void setCustomLayoutID(int layoutID) {
        this.customLayoutID = layoutID;
        RoomListActivity.Companion.setReplaceLayoutID(layoutID);
    }
}

File: liveroom-uikit/src/main/java/com/qlive/uikit/RoomListPage.kt
package com.qlive.uikit

import com.qlive.sdk.QPage

/**
 * Room list page
 */
class RoomListPage : QPage {
    /**
     * 自定义布局 如果需要替换自定义布局
     */
    var customLayoutID = -1
    set(value) {
        field = value
        RoomListActivity.replaceLayoutID = value
    }
}